lib/ronin/recon/builtin/web/screenshot.rb

```diff
@@ -0,0 +1,62 @@
+# frozen_string_literal: true
+#
+# ronin-recon - A micro-framework and tool for performing reconnaissance.
+#
+# Copyright (c) 2023-2024 Hal Brodigan (postmodern.mod3@gmail.com)
+#
+# ronin-recon is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License as published
+# by the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# ronin-recon is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with ronin-recon.  If not, see <https://www.gnu.org/licenses/>.
+#
+
+require 'ronin/recon/worker'
+
+require 'ronin/web/browser'
+
+module Ronin
+  module Recon
+    module Web
+      #
+      # A recon worker that takes a screenshot of a page.
+      #
+      class Screenshot < Worker
+        register 'web/screenshot'
+
+        summary 'Visits a website and takes a screenshot of it'
+        description <<~DESC
+          Visits a website and takes a screenshot of it
+        DESC
+
+        accepts URL
+        outputs URL # TODO: REMOVE
+
+        param :output_path, String, required: true,
+                                    desc:     'URL of the website you want to screenshot'
+
+        #
+        # Visits a website and takes a screenshot of it.
+        #
+        # @param [Values::URL] url
+        #   The URL of the website you want to screenshot.
+        #
+        def process(url)
+          Ronin::Web::Browser.open do |browser|
+            browser.go_to(url)
+            browser.screenshot(path: params[:output_path])
```

Comment:
I should add an extra keyword argument to `Ronin::Web::Browser::Agent#screenshot` to support passing in a root directory, so that the screenshot image path is derived from the page URL. This would be similar to the [logic in the `ronin-web screenshot` command](https://github.com/ronin-rb/ronin-web/blob/27dcb7b784baab4e4244d0daafa4d287134df927/lib/ronin/web/cli/commands/screenshot.rb#L175-L177). Maybe also return the screenshot image path.

Reply:
Hmm, in the mean time, maybe just adapt the code from `ronin-web screenshot`. You could also add a `form` param that defaults to `:png`, and replace `output_path` with `output_dir`. Then you could do something like:

```ruby
path = File.join(params[:output_dir],page.uri.host,page.uri.request_uri)
path << 'index' if path.end_with?('/')
path << ".#{params[:format]}"
```

Reply:
My mistake, `browser.page.url` which returns a String.